source/logic.rst

```diff
@@ -0,0 +1,492 @@
+.. _logic:
+
+Logic
+=====
+
+In the last chapter, we dealt with equations, inequalities,
+and basic mathematical statements like
+":math:`x` divides :math:`y`."
+Complex mathematical statements are built up from
+simple ones like these
+using logical terms like "and," "or," "not," and
+"if ... then," "every," and "some."
+In this chapter, we show you how to work with statements
+that are built up in this way.
+
+
+.. _using_implication_and_the_universal_quantifier:
+
+Implication and the Universal Quantifier
+----------------------------------------
+
+Consider the statement after the ``#check``:
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    #check ∀ x y ε : ℝ,
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε
+    -- END
+
+In words, we would say "for every ``x``, ``y``, and ``ε``,
+if ``0 < ε ≤ 1``, the absolute value of ``x`` is less than ``ε``,
+and the absolute value of ``y`` is less than ``ε``,
+then the absolute value of ``x * y`` is less than ``ε``."
+In Lean, the implication arrows associate to the right,
+making it easy to chain implications in this way.
+
+You have already seen that even though the universal quantifier
+in this statement
+ranges over objects and the implication arrows introduce hypotheses,
+Lean treats the two in very similar ways.
+In particular, if you have proved a theorem of that form,
+you can apply it to objects and hypotheses in the same way:
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ x y ε : ℝ,
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    sorry
+
+    section
+      variables a b δ : ℝ
+      variables (h₀ : 0 < δ) (h₁ : δ ≤ 1)
+      variables (ha : abs a < δ) (hb : abs b < δ)
+
+      #check my_lemma a b δ
+      #check my_lemma a b δ h₀ h₁
+      #check my_lemma a b δ h₀ h₁ ha hb
+    end
+    -- END
+
+You have also already seen that it is common in Lean
+to use curly brackets to make quantified variables implicit
+when they can be inferred from subsequent hypotheses.
+When we do that, we can just apply a lemma to the hypotheses without
+mentioning the objects.
+
+.. code-block:: lean
+
+    import data.real.basic
+
+    -- BEGIN
+    lemma my_lemma : ∀ {x y ε : ℝ},
+      0 < ε → ε ≤ 1 → abs x < ε → abs y < ε → abs (x * y) < ε :=
+    sorry
+
+    section
+      variables a b δ : ℝ
+      variables (h₀ : 0 < δ) (h₁ : δ ≤ 1)
+      variables (ha : abs a < δ) (hb : abs b < δ)
+
+      #check my_lemma h₀ h₁ ha hb
+    end
+    -- END
+
+At this stage, you also know that if you use
+the ``apply`` tactic to apply ``my_lemma``
+to a goal of the form ``abs (a * b) < δ``,
+you are left with new goals that require you to  prove
+each of the hypotheses.
+
+To prove a statement like this, use the ``intros`` tactic.
```

Comment:
Maybe mention sometimes that we still have to introduce `x y ε` even if they are implicit. Implicitness is about using the lemma, it doesn't change anything to the proof.